browser_use/browser/session.py

```diff
@@ -288,13 +292,13 @@ async def stop(self) -> None:
 
 		if self.browser_profile.keep_alive:
 			return  # nothing to do if keep_alive=True, leave the browser running
-
-		if self.browser_context or self.browser:
+		browser = self.browser_context and self.browser_context.browser or self.browser
+		if browser is not None:
 			try:
-				await (self.browser_context or self.browser).close()
+				await browser.close()
 				logger.info(
 					f'🛑 Stopped the {self.browser_profile.channel.name.lower()} browser '
-					f'keep_alive=False user_data_dir={_log_pretty_path(self.browser_profile.user_data_dir) or "<incognito>"} cdp_url={self.cdp_url or self.wss_url} pid={self.browser_pid}'
+					f'keep_alive=False user_data_dir={_log_pretty_path(Path(self.browser_profile.user_data_dir)) if self.browser_profile.user_data_dir else "<incognito>"} cdp_url={self.cdp_url or self.wss_url} pid={self.browser_pid}'
```

Comment:
not a big deal but this isn't technically necessary, `_log_pretty_path` returns `''` in case it's given a falsy path